### ISSUE TITLE
rxcouch with Angular 6-cli

### ISSUE THREAD

**Issue by dfulgham** [mkeen/rxcouch#1]
Trying to implement your library within an angular 6-cli test app.

getting the following error when trying to retrieve an existing document using _id string., can't seem to figure out where it is failing?

```
core.js:1673 ERROR TypeError: You provided 'undefined' where a stream was expected. You can provide an Observable, Promise, Array, or Iterable.
    at subscribeTo (subscribeTo.js:41)
    at subscribeToResult (subscribeToResult.js:6)
    at MergeMapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._innerSub (mergeMap.js:70)
    at MergeMapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._tryNext (mergeMap.js:67)
    at MergeMapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._next (mergeMap.js:50)
    at MergeMapSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at Observable._subscribe (couchdbdocumentcollection.js:31)
    at Observable.push../node_modules/rxjs/_esm5/internal/Observable.js.Observable._trySubscribe (Observable.js:42)
    at Observable.push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe (Observable.js:28)
    at MergeMapOperator.push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapOperator.call (mergeMap.js:28)
```

very simple test in a component, the document loads from the database correctly (rxhttp), but causes the above error.

```
import { Component, OnInit } from "@angular/core";
import { CouchDB } from "@mkeen/rxcouch";

@Component({
  selector: "app-root",
  templateUrl: "./app.component.html",
  styleUrls: ["./app.component.css"]
})
export class AppComponent implements OnInit {
  private rxCouch = new CouchDB("localhost", "rxcouch_test", 5984);

  ngOnInit() {
    this.rxCouch
      .doc("cf832aba6b1fb50997e5e26656002a5d")
      .subscribe(document => {
        console.log(document);
      });
  }
}

```
creating new documents works fine though.

**Comment by dfulgham** [comment]
@mkeen 
Don't know if this is a proper fix, but it looks like the following line in couchdb.ts line 236 

```
const Document = this.documents.doc(doc._id);
```

should be:

```
const Document = this.documents.doc(doc);
```

Which seems to be working with the change, able to initially load the existing document and is subscribed to future changes on the db.

However for some reason the changed function is being called after a change with an undefined document (couchdbdocumentcollection.js line 14).  A check should be added for undefined document.
```

ERROR TypeError: Cannot read property '_id' of undefined
    at CouchDBDocumentCollection.push../node_modules/@mkeen/rxcouch/dist/couchdbdocumentcollection.js.CouchDBDocumentCollection.changed (couchdbdocumentcollection.js:14)
    at SafeSubscriber._next (couchdb.js:48)
    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub (Subscriber.js:195)
    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next (Subscriber.js:133)
    at Subscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next (Subscriber.js:77)
    at Subscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at TakeUntilSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next (Subscriber.js:77)
    at TakeUntilSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at httprequest.js:173
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:388)
```

**Comment by dfulgham** [comment]
The error at couchdbdocumentcollection.js above is caused by the heartbeat or timeout being reach on the continuous feed to the CouchDB database.  Once the heartbeat (60sec) has been reached the database sends a empty response line and once timeout (also default to 60sec) has been reach the last line in the response indicates the last sequence (last-seq)  as seen below and then stops responding.

```

{"seq":"16-g1AAAACbeJzLYWBgYMpgTmEQTM4vTc5ISXLIyU9OzMnILy7JAUklMiTV____PyuDOZEnFyjAnpRkaGlgmIpNAx5j8liAJEMDkPoPNY0FbJqRkWVispEJNn1ZAHSAMKE","id":"cf832aba6b1fb50997e5e26656002a5d","changes":[{"rev":"11-7569a42de6429259bd183f36f73a94c7"}],"doc":{"_id":"cf832aba6b1fb50997e5e26656002a5d","_rev":"11-7569a42de6429259bd183f36f73a94c7","test":"test5"}}
               // empty response
{"seq":"17-g1AAAACbeJzLYWBgYMpgTmEQTM4vTc5ISXLIyU9OzMnILy7JAUklMiTV____PyuDOZE3FyjAnpRkaGlgmIpNAx5j8liAJEMDkPoPNY0FbJqRkWVispEJNn1ZAHTwMKI","id":"cf832aba6b1fb50997e5e26656002a5d","changes":[{"rev":"12-d50e5c4dc4d636842bb88e9e4eff3c9a"}],"doc":{"_id":"cf832aba6b1fb50997e5e26656002a5d","_rev":"12-d50e5c4dc4d636842bb88e9e4eff3c9a","test":"test6"}}
{"last_seq":"17-g1AAAACbeJzLYWBgYMpgTmEQTM4vTc5ISXLIyU9OzMnILy7JAUklMiTV____PyuDOZE3FyjAnpRkaGlgmIpNAx5j8liAJEMDkPoPNY0FbJqRkWVispEJNn1ZAHTwMKI","pending":0}

```

For this to work in production once the last-seq line is received we need to refresh the http request.

**Comment by mkeen** [comment]
Should be fixed by #3. Will verify.
